x/multisig/model.go

```diff
@@ -13,32 +13,42 @@ const (
 	SequenceName = "id"
 )
 
-// enforce that Contract fulfils desired interface compile-time
+// Weight represents the strength of a signature.
+type Weight uint32
+
+func (w Weight) Validate() error {
```

Comment:
Not really sure why we need a type for one function.
Fine to keep it here, but I'd almost expect more logic with this.

Reply:
I like it. A dedicated type is DDD style

Reply:
I did it for the [protobuf file to show that certain fields are about the same value](https://github.com/iov-one/weave/blob/65a530b5ca991468699edb2f85368f2dcf7246b4/x/multisig/codec.proto#L25). They are all of type `Weight` which makes you think about them as a single functionality.

I agree that using uint32 is less complex.

Do you think it is worth keeping the `Weight` type or this is an overkill for what it provides and using simple uint32 is enough?

Reply:
It's fine design.
We use custom type aliases for many different strings in iov-core for example.
But I hadn't seen such use in go/protobuf, mainly when needed to add functionality (like Address).

It's fine with me, just wanted to understand the reasoning